sigar_linux.go

```diff
@@ -137,6 +137,103 @@ func (self *CpuList) Get() error {
 
 	return err
 }
+func (self *NetProtoV6Stats) Get() error {
+	return readFile(Procd+"/net/snmp6", func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) != 2 {
+			return true
+		}
+		switch fields[0] {
+		case "Ip6InReceives":
+			self.IP.InReceives, _ = strtoull(fields[1])
+		case "Ip6InAddrErrors":
+			self.IP.InAddrErrors, _ = strtoull(fields[1])
+		case "Ip6OutForwDatagrams":
+			self.IP.ForwDatagrams, _ = strtoull(fields[1])
+		case "Ip6InDelivers":
+			self.IP.InDelivers, _ = strtoull(fields[1])
+		case "Ip6OutRequests":
+			self.IP.OutRequests, _ = strtoull(fields[1])
+		case "Ip6InDiscards":
+			self.IP.InDiscards, _ = strtoull(fields[1])
+		case "Icmp6InMsgs":
+			self.ICMP.InMsgs, _ = strtoull(fields[1])
+		case "Icmp6InErrors":
+			self.ICMP.InErrors, _ = strtoull(fields[1])
+		case "Icmp6OutDestUnreachs":
+			self.ICMP.InDestUnreachs, _ = strtoull(fields[1])
+		case "Icmp6OutMsgs":
+			self.ICMP.OutMsgs, _ = strtoull(fields[1])
+		case "Udp6InDatagrams":
+			self.UDP.InDatagrams, _ = strtoull(fields[1])
+		case "Udp6OutDatagrams":
+			self.UDP.OutDatagrams, _ = strtoull(fields[1])
+		case "Udp6InErrors":
+			self.UDP.InErrors, _ = strtoull(fields[1])
+		case "Udp6NoPorts":
+			self.UDP.NoPorts, _ = strtoull(fields[1])
+		}
+		return true
+	})
+}
+
+func (self *NetProtoV4Stats) Get() error {
+	return readFile(Procd+"/net/snmp", func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			return true
+		}
+
+		switch fields[0] {
+		case "Ip:":
+			if fields[1] == "Forwarding" || len(fields) < 20 {
```

Comment:
nit: should add comments in here saying that lines alternate between describing the following line, and the data itself.  And then this string being first signals it is a description line